bip-0339.mediawiki

```diff
@@ -43,7 +43,8 @@ announcing and fetching transactions.
 # The protocol version of nodes implementing this BIP must be set to 70016 or higher.
 # The WTXIDRELAY message may be sent in response to a VERSION message from a peer whose protocol version is >= 70016, and prior to sending a VERACK. A WTXIDRELAY message received after a VERACK message MUST be ignored or treated as invalid.
 # A new inv type MSG_WTX (0x00000005) is added, for use in both INV messages and GETDATA requests, indicating that the hash being referenced is a transaction's wtxid. In the case of GETDATA requests, MSG_WTX implies that the transaction being requested should be serialized with witness as well, as described in BIP 144.
-# After a node has sent and received a WTXIDRELAY message to/from a given peer, the node is required to use the MSG_WTX inv-type when announcing transactions to that peer, or requesting announced transactions from that peer.
+# After a node has received a WTXIDRELAY message from a peer, the node SHOULD use the MSG_WTX inv-type when announcing transactions to that peer.
```

Comment:
> Why the relaxation to SHOULD here?

Sorry, you're right. This should stay as MUST instead of SHOULD.

> Also, why delete the requirement to request announced transactions from that peer using MSG_WTX getdata messages?

I think with your improved sentence below, this isn't needed here (and actually makes things less clear). I also think it's better to document the announce behaviour and the request behaviour in different bullet points.